cmd/process-agent/main_common.go

```diff
@@ -74,8 +74,17 @@ var (
 )
 
 func getSettingsClient() (settings.Client, error) {
-	// Set up the config in case the cmd_port was specified
-	_, err := config.NewAgentConfig(loggerName, opts.configPath, opts.sysProbeConfigPath)
+	// Set up the config so we can get the port later
+	// We set this up differently from the main process-agent because this way is quieter
+	cfg := config.NewDefaultAgentConfig(false)
```

Comment:
Just as a curiosity, why do we need to call this function here? Isn't the `cmd_port` setting fetched with ?
```
ddconfig.Datadog.GetInt("process_config.cmd_port")
```

Reply:
It is but ddconfig.Datadog first has to be initialized. To do this we:
1. Initialize an AgentConfig. This gives us access to it's receivers which will allow us to load in environment variables.
2. If a config file is specified, we tell ddconfig.Datadog to use that one instead.
3. Use the AgentConfig to initialize environment variables. This has to be done this way because config.loadEnvVariables is not exported, and is only called from AgentConfig.LoadProcessYamlConfig()

This essentially loads process_config.port with this priority:
1. Load from defaults
2. Override from config file if set
3. Override from environment variables if set.

Its unfortunate how complicated this has to be, but I definitely think it's possible to improve this process in the future.

Reply:
Looking at this now though I realize I do change the global ddconfig. This is an old change from [this file](https://github.com/DataDog/datadog-agent/pull/8693/files#diff-131d9407844f0970869481e7d1c542bb08006b7df86de076ae20fc64bb5366bc) and it was when I was trying to figure out how to set up the config.

Reply:
That makes sense! Thanks for clarifying it.